File: model/message.go
package model

import (
	"encoding/json"
)

// Event is the messages send over the websocket
// Used to differ between different actions
type Event struct {
	Type    string          `json:"type"`
	Payload json.RawMessage `json:"payload"`
}

const (
	SendMessageEvent  = "send_message"
	ErrorMessageEvent = "error_message"
)

// The payload sent in the SendMessageEvent
type SendMessagePayload struct {
	Room      string `json:"room"`
	User      string `json:"user"`
	Message   string `json:"message"`
	CreatedAt string `json:"createdAt"`
}

File: frontend/src/model/message.ts
// Use both for send messages to other client and to receive message fromother client
export interface Message {
  room: string;
  user: string;
  message: string;
  createdAt: string;
}

// return new Message
export function NewMessage(
  room: string,
  user: string,
  message: string
): Message {
  return {
    room: room,
    user: user,
    message: message,
    createdAt: Date.now().toString(),
  };
}
